### PR TITLE
[release-v1.11] sync upstream deps

### PR DESCRIPTION
**Pull request by matzew**
Basically, this PR runs:
```
 ./hack/update-deps.sh --upgrade --release 1.11
```

### CODE REVIEW

File: vendor/knative.dev/pkg/controller/stats_reporter.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func (r *reporter) ReportReconcile(duration time.Duration, success string, key t
 		return err
 	}
 
-	// TODO skonto: fix latency histogram
-	metrics.RecordBatch(ctx, reconcileCountStat.M(1))
+	metrics.RecordBatch(ctx, reconcileCountStat.M(1),
+		reconcileLatencyStat.M(duration.Milliseconds()))
```
Comment (pierDipi) [inline vendor/knative.dev/pkg/controller/stats_reporter.go:203]: Some missing patches to vendor

Comment (matzew) [reply vendor/knative.dev/pkg/controller/stats_reporter.go:203]: MAKE RELEASE=1.11 is not giving me any diff

**Comment by matzew** [comment]
The outdated patches were _not_ on this branch.

Re-applied the vendor patches (see commit message)

File: vendor/knative.dev/reconciler-test/pkg/eventshub/resources.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ func Install(name string, options ...EventsHubOption) feature.StepFn {
 			// No event recording desired, just logging.
 			envs[EventLogsEnv] = "logger"
 			cfg["envs"] = envs
-			cfg["sink"] = sinkURL
```
Comment (matzew) [inline vendor/knative.dev/reconciler-test/pkg/eventshub/resources.go:159]: @pierDipi  this update comes in via 

https://github.com/knative-extensions/reconciler-test/commit/03cc77c11831ef010415c5bf6ecb38abf6b6780e

**Comment by matzew** [comment]
Fails on `TestApiServerSourceValidationWebhookConfigurationOnCreate` 

Does that ring a bell?!

**Comment by creydr** [comment]
yes. It's the same 
```
        	Error:      	Error message not equal:
        	            	expected: "admission webhook \"validation.webhook.eventing.knative.dev\" denied the request: validation failed: invalid value: Unknown: spec.mode"
        	            	actual  : "failed to create resource admission webhook \"validation.webhook.eventing.knative.dev\" denied the request: validation failed: invalid value: Unknown: spec.mode - Resource:\napiVersion: sources.knative.dev/v1\nkind: ApiServerSource\nmetadata:\n  name: apiserversource-cwoihclx\n  namespace: test-cmrarmsg\nspec:\n  mode: Unknown\n  resources:\n    - apiVersion: v1\n      kind: Event\n  serviceAccountName: foo-sa\n  sink:\n    ref:\n      apiVersion: v1\n      kind: Service\n      name: foo-svc\n      namespace: null\n"
```

My u/s PR included this fix: https://github.com/knative/eventing/pull/7362. Did you take this as a base?

**Comment by matzew** [comment]
/hold
lets see if all really passes, now 
I will than unhold myself

**Review by creydr — COMMENTED** [review]
/lgtm

**Pull request merged**